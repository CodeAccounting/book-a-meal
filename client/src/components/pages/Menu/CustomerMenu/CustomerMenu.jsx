import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import MealCard from '../../../shared/MealCard';
import { userPropTypes, mealObjPropTypes } from '../../../../helpers/proptypes';
import CustomerView from '../../../shared/CustomerView';

/**
 * @exports
 * @class CustomerMenu
 * @extends Component
 * @classdesc Creates CustomerMenu Component
 * @returns {JSX} CustomerMenu Component
 */
class CustomerMenu extends Component {
  static propTypes = {
    ...userPropTypes,
    meals: PropTypes.arrayOf(mealObjPropTypes).isRequired,
    isFetching: PropTypes.bool.isRequired,
    logout: PropTypes.func.isRequired,
    fetchMenu: PropTypes.func.isRequired,
  }

  /**
   * @memberof CustomerMenu
   * @returns {JSX} CustomerMenu Component
   */
  componentDidMount() {
    this.fetchMenu();
  }

  /**
   * @memberof CustomerMenu
   * @returns {JSX} CustomerMenu Component
  */
  fetchMenu = () => {
    this.props.fetchMenu(this.props.currentDay);
  }

  /**
   * @memberof CustomerMenu
   * @returns {JSX} CustomerMenu Component
  */
  renderMenu = () => {
    const { meals } = this.props;
    return (
      <Fragment>
        {meals.length === 0 && <p className="text-center">{'There are no Meals on Today\'s Menu'}</p>}
        {meals.length !== 0 &&
          <div className="card-group meals-wrapper" id="card-group">
            {meals.map(meal =>
              (<MealCard
                type="customer"
                key={meal.id}
                meal={meal}
              />))}
          </div>}
      </Fragment>
    );
  }

  /**
   * @memberof CustomerMenu
   * @returns {JSX} CustomerMenu Component
  */
  render() {
    const { user, logout, isFetching } = this.props;

    return (
      <CustomerView
        user={user}
        logout={logout}
        type="menu"
        isFetching={isFetching}
      >
        <div className="content-wrapper meals user-meals">
          <div className="page-heading">
            <h2>{'Today\'s Menu'}</h2>
            <hr />
          </div>
          {this.renderMenu()}
        </div>
      </CustomerView>
    );
  }
}

export default CustomerMenu;
