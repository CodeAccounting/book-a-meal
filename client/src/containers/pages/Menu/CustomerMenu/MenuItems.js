import { connect } from 'react-redux';
import MenuItems from '../../../../components/pages/Menu/CustomerMenu/MenuItems';

const mapStateToProps = state => ({
  isFetching: state.isFetching,
  meals: state.menu.meals,
  metadata: state.menu.metadata
});

export default connect(mapStateToProps)(MenuItems);
