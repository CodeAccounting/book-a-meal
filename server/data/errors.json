{
  "401": "Unauthorized",
  "403": "Forbidden",
  "404": "Not Found",
  "422": "Validation Error",
  "500": "Internal Server Error",
  "empty": "Empty PUT Requests Are Not Allowed"
}