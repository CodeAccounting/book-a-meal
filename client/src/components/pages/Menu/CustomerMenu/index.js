import { connect, } from 'react-redux';
import { bindActionCreators } from 'redux';
import CustomerMenu from './CustomerMenu';
import { logout } from '../../../../store/operations/auth';
import { fetchMenu } from '../../../../store/operations/menu';

const mapStateToProps = state => ({
  isFetching: state.isFetching,
  meals: state.menu.meals,
  metadata: state.meals.metadata
});

const mapDispatchToProps = dispatch => bindActionCreators({
  logout, fetchMenu
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(CustomerMenu);
