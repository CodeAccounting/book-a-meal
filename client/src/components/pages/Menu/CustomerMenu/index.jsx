import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import MealCard from '../../../shared/MealCard';
import Cart from '../../../shared/Cart';
import { userPropTypes, mealObjPropTypes } from '../../../../helpers/proptypes';
import View from '../../../shared/View';
import Notification from '../../../shared/Notification';
import checkShopOpen from '../../../../helpers/checkShopOpen';
import getOrderFromLocalStorage from '../../../../helpers/getOrderFromLocalStorage';
import updateLocalStorageOrder from '../../../../helpers/updateLocalStorageOrder';
import CardGroup from '../../../shared/CardGroup';
import './CustomerMenu.scss';

/**
 * @exports
 * @class CustomerMenu
 * @extends Component
 * @classdesc Creates CustomerMenu Component
 * @returns {JSX} CustomerMenu Component
 */
class CustomerMenu extends Component {
  static propTypes = {
    ...userPropTypes,
    meals: PropTypes.arrayOf(mealObjPropTypes).isRequired,
    isFetching: PropTypes.bool.isRequired,
    logout: PropTypes.func.isRequired,
    fetchMenu: PropTypes.func.isRequired,
  }

  /**
   * @constructor
   * @memberof CustomerMenu
   * @param {object} props
   * @returns {JSX} CustomerMenu Component
   */
  constructor(props) {
    super();

    this.state = {
      order: getOrderFromLocalStorage(props.user)
    };
  }

  /**
   * @memberof CustomerMenu
   * @returns {JSX} CustomerMenu Component
   */
  componentDidMount() {
    this.fetchMenu();
  }

  /**
   * @memberof CustomerMenu
   * @param {string} itemId
   * @returns {void}
   */
  getItemIndex = itemId => this.state.order.meals.findIndex(item => item.id === itemId)

  /**
   * @memberof CustomerMenu
   * @param {object} meal
   * @returns {func} addOrderItem|updateOrderItem
   */
  handleOrderMealClick = (meal) => {
    const itemIndex = this.getItemIndex(meal.id);

    if (itemIndex !== -1) {
      const newQuantity = this.state.order.meals[itemIndex].quantity + 1;

      const orderItem = this.changeOrderQuantity(meal.id, newQuantity);

      return this.updateOrderItem(orderItem);
    }

    return this.addOrderItem(meal);
  }

  /**
   * @memberof CustomerMenu
   * @param {object} meal
   * @returns {void}
   */
  addOrderItem = (meal) => {
    const item = {
      id: meal.id,
      title: meal.title,
      price: meal.price,
      quantity: 1
    };

    this.setState(prevState => ({
      order: {
        ...prevState.order,
        meals: [...prevState.order.meals, item]
      }
    }), () =>
      updateLocalStorageOrder(this.props.user.id, this.state.order));
  }

  /**
   * @memberof CustomerMenu
   * @param {object} orderItem
   * @returns {void}
   */
  updateOrderItem = (orderItem) => {
    const itemIndex = this.getItemIndex(orderItem.id);

    this.setState(prevState => ({
      order: {
        ...prevState.order,
        meals: [
          ...prevState.order.meals.slice(0, itemIndex),
          {
            ...prevState.order.meals[itemIndex],
            ...orderItem
          },
          ...prevState.order.meals.slice(itemIndex + 1)
        ]
      }
    }), () =>
      updateLocalStorageOrder(this.props.user.id, this.state.order));
  }

  /**
   * @memberof CustomerMenu
   * @param {string} itemId
   * @returns {void}
   */
  removeOrderItem = (itemId) => {
    this.setState(prevState => ({
      order: {
        ...prevState.order,
        meals: prevState.order.meals.filter(orderItem => orderItem.id !== itemId)
      }
    }), () =>
      updateLocalStorageOrder(this.props.user.id, this.state.order));
  }

  /**
   * @memberof CustomerMenu
   * @param {object} event
   * @param {string} itemId
   * @returns {func} updateOrderItem
   */
  handleQuantityInputChange = (event, itemId) => {
    const quantity = event.target.value <= 0 ? 1 : event.target.value;

    const orderItem = this.changeOrderQuantity(itemId, parseInt(quantity, 10));
    return this.updateOrderItem(orderItem);
  }

  /**
   * @memberof CustomerMenu
   * @param {string} itemId
   * @param {number} quantity
   * @returns {object} orderItem
   */
  changeOrderQuantity = (itemId, quantity) => {
    const orderItem = this.state.order.meals.find(item => item.id === itemId);

    orderItem.quantity = quantity;

    return orderItem;
  }

  /**
   * @memberof CustomerMenu
   * @param {object} metadata
   * @returns {func} fetchMenu
  */
  fetchMenu = metadata => this.props.fetchMenu(this.props.currentDay, metadata);

  /**
   * @memberof CustomerMenu
   * @param {object} metadata
   * @returns {func} load more menu
  */
  loadMoreMenu = () => this.fetchMenu(this.props.metadata)

  /**
   * @memberof CustomerMenu
   * @returns {JSX} CustomerMenu Component
  */
  renderMenu = () => {
    const { meals } = this.props;
    const menu = meals.map(meal =>
      (<MealCard
        type="customer"
        key={meal.id}
        meal={meal}
        orderMeal={() => this.handleOrderMealClick(meal)}
        inBasket={!!this.state.order.meals.find(item => item.id === meal.id)}
      />));

    return (
      <Fragment>
        {meals.length === 0 && <p className="text-center">{'There are no Meals on Today\'s Menu'}</p>}
        {meals.length !== 0 &&
          <CardGroup
            items={menu}
            metadata={this.props.metadata}
            loadMore={this.loadMoreMenu}
          />}
      </Fragment>
    );
  }

  /**
   * @memberof CustomerMenu
   * @returns {JSX} CustomerMenu Component
  */
  render() {
    const { user, logout, isFetching } = this.props;
    const isShopOpen = checkShopOpen();

    return (
      <View user={user} logout={logout} type="menu" isFetching={isFetching}>
        <div className="meals user-meals">
          <div className="user-menu">
            <div className="main-menu">
              <div className="page-heading">
                <h2>{'Today\'s Menu'}</h2>
                <hr />
              </div>
              {!isShopOpen && <Notification message="Ordering is only available between 8:30am and 4:00pm. Please check back later." />}
              {this.renderMenu()}
            </div>
            <Cart
              order={this.state.order.meals}
              handleQuantityInputChange={this.handleQuantityInputChange}
              removeOrderItem={this.removeOrderItem}
            />
          </div>
        </div>
      </View>
    );
  }
}

export default CustomerMenu;
