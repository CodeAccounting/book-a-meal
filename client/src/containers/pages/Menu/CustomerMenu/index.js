import { connect, } from 'react-redux';
import { bindActionCreators } from 'redux';
import CustomerMenu from '../../../../components/pages/Menu/CustomerMenu';
import { logout } from '../../../../actions/auth';
import { fetchMenu } from '../../../../actions/menu';

const mapDispatchToProps = dispatch => bindActionCreators({
  logout, fetchMenu
}, dispatch);

export default connect(null, mapDispatchToProps)(CustomerMenu);
