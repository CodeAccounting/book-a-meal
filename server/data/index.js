import users from './users.json';
import meals from './meals.json';
import menu from './menu.json';
import orders from './orders.json';
import errors from './errors.json';

export default {
  users, meals, menu, orders, errors
};
