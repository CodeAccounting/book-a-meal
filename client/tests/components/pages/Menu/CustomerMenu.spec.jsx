import React from 'react';
import moment from 'moment';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import CustomerMenu from '../../../../src/components/pages/Menu/CustomerMenu/CustomerMenu';
import ConnectedCustomerMenu from '../../../../src/components/pages/Menu/CustomerMenu';
import { customer, caterersMealsObj, initialValues } from '../../../setup/data';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({
  ...initialValues, menu: { ...initialValues.menu, meals: caterersMealsObj.meals }
});
const { now } = Date;
const currentDay = moment().format('YYYY-MM-DD');

describe('CustomerMenu', () => {
  beforeAll(() => {
    Date.now = jest.fn(() => new Date(currentDay).getTime() + (60 * 60 * 13 * 1000));
  });

  afterAll(() => {
    Date.now = now;
  });

  it('renders correctly when not fetching', () => {
    const shallowWrapper = shallow(<CustomerMenu
      user={customer}
      logout={jest.fn()}
      fetchMenu={jest.fn()}
      {...caterersMealsObj}
      isFetching={false}
    />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.find('MealCard')).toBeTruthy();
  });

  it('renders Preloader when fetching', () => {
    const shallowWrapper = shallow(<CustomerMenu
      user={customer}
      logout={jest.fn()}
      fetchMenu={jest.fn()}
      {...caterersMealsObj}
      isFetching
    />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.find('Preloader')).toBeTruthy();
  });

  it('renders message when not fetching and there are no meals on the menu', () => {
    const shallowWrapper = shallow(<CustomerMenu
      user={customer}
      logout={jest.fn()}
      fetchMenu={jest.fn()}
      meals={[]}
      isFetching
    />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.find('p').text()).toEqual('There are no Meals on Today\'s Menu');
  });

  it('renders connected component', () => {
    const comp = (
      <Provider store={store}>
        <ConnectedCustomerMenu
          user={customer}
          {...caterersMealsObj}
        />
      </Provider>);

    const wrapper = mount(comp, rrcMock.get());

    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });

  it('renders message when shop is closed', () => {
    const dateNowSpy = jest.spyOn(Date, 'now').mockImplementation(() => new Date(currentDay).getTime() + (60 * 60 * 18 * 1000));
    const wrapper = mount(<CustomerMenu
      user={customer}
      logout={jest.fn()}
      fetchMenu={jest.fn()}
      isFetching={false}
      {...caterersMealsObj}
    />, rrcMock.get());

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('p.notif').text()).toEqual('Ordering is not available between 4:00pm and 8:30am. Please check back later.');
    dateNowSpy.mockReset();
    dateNowSpy.mockRestore();
  });

  describe('Order Creation', () => {
    beforeAll(() => {
      Date.now = jest.fn(() => new Date(currentDay).getTime() + (60 * 60 * 13 * 1000));

      localStorage.setItem('bookamealorder', JSON.stringify({
        userId: 'a09a5570-a3b2-4e21-94c3-5cf483dbd1ac',
        order: [{
          id: '81211c24-51c0-46ec-b1e0-18db55880958',
          title: 'Jollof Rice, Beef and Plantain',
          price: '1500.00',
          quantity: 1
        }],
        date: moment().format('YYYY-MM-DD')
      }));
    });

    afterAll(() => {
      Date.now = now;
    });

    it('adds an order', () => {
      const comp = (
        <CustomerMenu
          user={customer}
          logout={jest.fn()}
          fetchMenu={jest.fn()}
          isFetching={false}
          {...caterersMealsObj}
        />);


      const wrapper = mount(comp, rrcMock.get());
      const addOrderSpy = jest.spyOn(wrapper.instance(), 'addOrderItem');
      const handleOrderMealClickSpy = jest.spyOn(wrapper.instance(), 'handleOrderMealClick');

      wrapper.find('.meal-card-btn').at(1).simulate('click');

      expect(handleOrderMealClickSpy).toHaveBeenCalled();
      expect(addOrderSpy).toHaveBeenCalled();
      expect(wrapper.state().order[1].title).toEqual('Vegetable Sharwama and Guava Smoothie');
      expect(wrapper.state().order[1].quantity).toEqual(1);

      wrapper.unmount();
    });

    it('updates orderItem on reorder', () => {
      const comp = (
        <CustomerMenu
          user={customer}
          logout={jest.fn()}
          fetchMenu={jest.fn()}
          isFetching={false}
          {...caterersMealsObj}
        />);

      const wrapper = mount(comp, rrcMock.get());
      const updateOrderItemSpy = jest.spyOn(wrapper.instance(), 'updateOrderItem');
      const handleOrderMealClickSpy = jest.spyOn(wrapper.instance(), 'handleOrderMealClick');

      wrapper.find('.meal-card-btn').at(0).simulate('click');

      expect(updateOrderItemSpy).toHaveBeenCalled();
      expect(handleOrderMealClickSpy).toHaveBeenCalled();
      expect(wrapper.state().order[0].quantity).toEqual(2);

      wrapper.unmount();
    });

    it('changes order quantity on input change', () => {
      const comp = (
        <CustomerMenu
          user={customer}
          logout={jest.fn()}
          fetchMenu={jest.fn()}
          isFetching={false}
          {...caterersMealsObj}
        />);

      const wrapper = mount(comp, rrcMock.get());
      const changeOrderQuantitySpy = jest.spyOn(wrapper.instance(), 'changeOrderQuantity');

      wrapper.find('input').at(0).simulate('change', { target: { value: 3 } });

      expect(changeOrderQuantitySpy).toHaveBeenCalled();
      expect(wrapper.state().order[0].quantity).toEqual(3);

      wrapper.unmount();
    });

    it('sets quantity input back to one if set to 0 or negative value', () => {
      const comp = (
        <CustomerMenu
          user={customer}
          logout={jest.fn()}
          fetchMenu={jest.fn()}
          isFetching={false}
          {...caterersMealsObj}
        />);

      const wrapper = mount(comp, rrcMock.get());

      wrapper.find('input').at(0).simulate('change', { target: { value: 0 } });
      expect(wrapper.state().order[0].quantity).toEqual(1);

      wrapper.find('input').at(0).simulate('change', { target: { value: -8 } });
      expect(wrapper.state().order[0].quantity).toEqual(1);

      wrapper.unmount();
    });

    it('sets empty array as state order when local storage date is not today', () => {
      const comp = (
        <CustomerMenu
          user={customer}
          logout={jest.fn()}
          fetchMenu={jest.fn()}
          isFetching={false}
          {...caterersMealsObj}
        />);

      localStorage.setItem('bookamealorder', JSON.stringify({
        userId: 'a09a5570-a3b2-4e21-94c3-5cf483dbd1ac',
        order: [{
          id: '81211c24-51c0-46ec-b1e0-18db55880958',
          title: 'Jollof Rice, Beef and Plantain',
          price: '1500.00',
          quantity: 1
        }],
        date: '2017-06-20'
      }));

      const wrapper = mount(comp, rrcMock.get());

      expect(wrapper.state().order.length).toEqual(0);

      wrapper.unmount();
    });

    it('removes orderItem when delete icon is click', () => {
      const comp = (
        <CustomerMenu
          user={customer}
          logout={jest.fn()}
          fetchMenu={jest.fn()}
          isFetching={false}
          {...caterersMealsObj}
        />);

      localStorage.setItem('bookamealorder', JSON.stringify({
        userId: 'a09a5570-a3b2-4e21-94c3-5cf483dbd1ac',
        order: [{
          id: '81211c24-51c0-46ec-b1e0-18db55880958',
          title: 'Jollof Rice, Beef and Plantain',
          price: '1500.00',
          quantity: 1
        }],
        date: moment().format('YYYY-MM-DD')
      }));

      const wrapper = mount(comp, rrcMock.get());

      wrapper.find('button.remove-order').simulate('click');

      expect(wrapper.state().order.length).toEqual(0);

      wrapper.unmount();
    });
  });
});
