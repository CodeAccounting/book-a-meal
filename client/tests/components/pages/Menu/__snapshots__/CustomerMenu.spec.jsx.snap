// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders Preloader when fetching 1`] = `
<CustomerView
  isFetching={true}
  logout={[MockFunction]}
  type="menu"
  user={
    Object {
      "email": "emiola@olasanmi.com",
      "firstname": "Emiola",
      "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
      "lastname": "Olasanmi",
      "role": "customer",
    }
  }
>
  <div
    className="content-wrapper meals user-meals"
  >
    <div
      className="page-heading"
    >
      <h2>
        Today's Menu
      </h2>
      <hr />
    </div>
    <React.Fragment>
      <div
        className="card-group meals-wrapper"
        id="card-group"
      >
        <MealCard
          key="81211c24-51c0-46ec-b1e0-18db55880958"
          meal={
            Object {
              "description": "Jollof Rice, Beef and Plantain. 2 pieces of beef per plate",
              "id": "81211c24-51c0-46ec-b1e0-18db55880958",
              "imageURL": "http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg",
              "price": "1500.00",
              "title": "Jollof Rice, Beef and Plantain",
              "vegetarian": false,
            }
          }
          toggleModal={null}
          type="customer"
        />
        <MealCard
          key="36d525d1-efc9-4b75-9999-3e3d8dc64ce3"
          meal={
            Object {
              "description": "Sharwama contains no animal products, perfect for dieters",
              "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
              "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
              "price": "1200.00",
              "title": "Vegetable Sharwama and Guava Smoothie",
              "vegetarian": true,
            }
          }
          toggleModal={null}
          type="customer"
        />
      </div>
    </React.Fragment>
  </div>
</CustomerView>
`;

exports[`Menu renders connected component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(CustomerMenu)
    meals={
      Array [
        Object {
          "description": "Jollof Rice, Beef and Plantain. 2 pieces of beef per plate",
          "id": "81211c24-51c0-46ec-b1e0-18db55880958",
          "imageURL": "http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg",
          "price": "1500.00",
          "title": "Jollof Rice, Beef and Plantain",
          "vegetarian": false,
        },
        Object {
          "description": "Sharwama contains no animal products, perfect for dieters",
          "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
          "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
          "price": "1200.00",
          "title": "Vegetable Sharwama and Guava Smoothie",
          "vegetarian": true,
        },
      ]
    }
    user={
      Object {
        "email": "emiola@olasanmi.com",
        "firstname": "Emiola",
        "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
        "lastname": "Olasanmi",
        "role": "customer",
      }
    }
  >
    <CustomerMenu
      fetchMenu={[Function]}
      isFetching={false}
      logout={[Function]}
      meals={
        Array [
          Object {
            "description": "Jollof Rice, Beef and Plantain. 2 pieces of beef per plate",
            "id": "81211c24-51c0-46ec-b1e0-18db55880958",
            "imageURL": "http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg",
            "price": "1500.00",
            "title": "Jollof Rice, Beef and Plantain",
            "vegetarian": false,
          },
          Object {
            "description": "Sharwama contains no animal products, perfect for dieters",
            "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
            "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
            "price": "1200.00",
            "title": "Vegetable Sharwama and Guava Smoothie",
            "vegetarian": true,
          },
        ]
      }
      user={
        Object {
          "email": "emiola@olasanmi.com",
          "firstname": "Emiola",
          "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
          "lastname": "Olasanmi",
          "role": "customer",
        }
      }
    >
      <CustomerView
        isFetching={false}
        logout={[Function]}
        type="menu"
        user={
          Object {
            "email": "emiola@olasanmi.com",
            "firstname": "Emiola",
            "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
            "lastname": "Olasanmi",
            "role": "customer",
          }
        }
      >
        <div
          className="admin user"
        >
          <Header
            active="menu"
            dateType={null}
            logout={[Function]}
            showTime={true}
            type="customer"
            user={
              Object {
                "email": "emiola@olasanmi.com",
                "firstname": "Emiola",
                "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                "lastname": "Olasanmi",
                "role": "customer",
              }
            }
          >
            <div
              className="header"
            >
              <div
                className="navbar white-navbar"
              >
                <CustomerHeader
                  active="menu"
                  dateType={null}
                  logout={[Function]}
                  showTime={true}
                  type="customer"
                  user={
                    Object {
                      "email": "emiola@olasanmi.com",
                      "firstname": "Emiola",
                      "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                      "lastname": "Olasanmi",
                      "role": "customer",
                    }
                  }
                >
                  <div
                    className="page-title"
                  >
                    <h3>
                      <a
                        href="/"
                      >
                        BOOK-A-MEAL
                      </a>
                    </h3>
                  </div>
                  <div
                    className="navlinks"
                  >
                    <Link
                      className="link"
                      clickHandler={null}
                      href="#"
                      id={null}
                    >
                      <button
                        className="link-btn link"
                        id={null}
                        onClick={null}
                      >
                        <img
                          alt="cart"
                          src="/img/cart.png"
                        />
                      </button>
                    </Link>
                    <div
                      className="user-info"
                    >
                      <Dropdown
                        content={
                          <UNDEFINED>
                            <a
                              className="active"
                              href="/"
                            >
                              Menu
                            </a>
                            <a
                              href="/"
                            >
                              Order History
                            </a>
                            <a
                              href="/"
                              onClick={[Function]}
                            >
                              Logout
                            </a>
                          </UNDEFINED>
                        }
                        toggler={
                          <div
                            className="username-circle circle-click"
                          >
                            <p
                              className="circle-click"
                            >
                              E
                              ▿
                            </p>
                          </div>
                        }
                        type="customer"
                      >
                        <div
                          className="dropdown"
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                        >
                          <Link
                            className="link"
                            clickHandler={[Function]}
                            id="dropdown-toggler"
                          >
                            <button
                              className="link-btn link"
                              id="dropdown-toggler"
                              onClick={[Function]}
                            >
                              <div
                                className="username-circle circle-click"
                              >
                                <p
                                  className="circle-click"
                                >
                                  E
                                  ▿
                                </p>
                              </div>
                            </button>
                          </Link>
                          <div
                            className="dropdown-content"
                            id="dropdown-content"
                          >
                            <a
                              className="active"
                              href="/"
                            >
                              Menu
                            </a>
                            <a
                              href="/"
                            >
                              Order History
                            </a>
                            <a
                              href="/"
                              onClick={[Function]}
                            >
                              Logout
                            </a>
                          </div>
                        </div>
                      </Dropdown>
                      <p
                        className="hide d-lg"
                      >
                        Emiola
                      </p>
                    </div>
                  </div>
                </CustomerHeader>
              </div>
            </div>
          </Header>
          <div
            className="content"
          >
            <div
              className="content-wrapper"
              id="has-modal"
            >
              <div
                className="content-wrapper meals user-meals"
              >
                <div
                  className="page-heading"
                >
                  <h2>
                    Today's Menu
                  </h2>
                  <hr />
                </div>
                <div
                  className="card-group meals-wrapper"
                  id="card-group"
                >
                  <MealCard
                    key="81211c24-51c0-46ec-b1e0-18db55880958"
                    meal={
                      Object {
                        "description": "Jollof Rice, Beef and Plantain. 2 pieces of beef per plate",
                        "id": "81211c24-51c0-46ec-b1e0-18db55880958",
                        "imageURL": "http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg",
                        "price": "1500.00",
                        "title": "Jollof Rice, Beef and Plantain",
                        "vegetarian": false,
                      }
                    }
                    toggleModal={null}
                    type="customer"
                  >
                    <div
                      className="meal-card order-meal-card"
                      id="meal-card"
                    >
                      <div
                        className="meal-card-header"
                      >
                        <img
                          alt="meal"
                          src="http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg"
                        />
                        <div
                          className="menu-card-title"
                        >
                          <p>
                            Jollof Rice, Beef and Plantain
                          </p>
                        </div>
                      </div>
                      <div
                        className="meal-card-body"
                      >
                        <div>
                          <h3>
                            ₦ 
                            1500.00
                          </h3>
                          <p>
                            Jollof Rice, Beef and Plantain. 2 pieces of beef per plate
                          </p>
                        </div>
                        <div
                          className="meal-card-action"
                        >
                          <button
                            className="btn btn-sec meal-card-btn"
                          >
                            Click to Order
                          </button>
                        </div>
                      </div>
                    </div>
                  </MealCard>
                  <MealCard
                    key="36d525d1-efc9-4b75-9999-3e3d8dc64ce3"
                    meal={
                      Object {
                        "description": "Sharwama contains no animal products, perfect for dieters",
                        "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
                        "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
                        "price": "1200.00",
                        "title": "Vegetable Sharwama and Guava Smoothie",
                        "vegetarian": true,
                      }
                    }
                    toggleModal={null}
                    type="customer"
                  >
                    <div
                      className="meal-card order-meal-card"
                      id="meal-card"
                    >
                      <div
                        className="meal-card-header"
                      >
                        <img
                          alt="meal"
                          src="https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350"
                        />
                        <div
                          className="menu-card-title"
                        >
                          <p>
                            Vegetable Sharwama and Guava Smoothie
                          </p>
                        </div>
                      </div>
                      <div
                        className="meal-card-body"
                      >
                        <div>
                          <h3>
                            ₦ 
                            1200.00
                          </h3>
                          <p>
                            Sharwama contains no animal products, perfect for dieters
                          </p>
                        </div>
                        <div
                          className="meal-card-action"
                        >
                          <button
                            className="btn btn-sec meal-card-btn"
                          >
                            Click to Order
                          </button>
                        </div>
                      </div>
                    </div>
                  </MealCard>
                </div>
              </div>
            </div>
          </div>
          <Footer>
            <div
              className="footer"
            >
              <p>
                © 2018. Iveren Shaguy. All images were gotten from Google and are not mine.
              </p>
            </div>
          </Footer>
        </div>
      </CustomerView>
    </CustomerMenu>
  </Connect(CustomerMenu)>
</Provider>
`;

exports[`Menu renders correctly when not fetching 1`] = `
<CustomerView
  isFetching={false}
  logout={[MockFunction]}
  type="menu"
  user={
    Object {
      "email": "emiola@olasanmi.com",
      "firstname": "Emiola",
      "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
      "lastname": "Olasanmi",
      "role": "customer",
    }
  }
>
  <div
    className="content-wrapper meals user-meals"
  >
    <div
      className="page-heading"
    >
      <h2>
        Today's Menu
      </h2>
      <hr />
    </div>
    <React.Fragment>
      <div
        className="card-group meals-wrapper"
        id="card-group"
      >
        <MealCard
          key="81211c24-51c0-46ec-b1e0-18db55880958"
          meal={
            Object {
              "description": "Jollof Rice, Beef and Plantain. 2 pieces of beef per plate",
              "id": "81211c24-51c0-46ec-b1e0-18db55880958",
              "imageURL": "http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg",
              "price": "1500.00",
              "title": "Jollof Rice, Beef and Plantain",
              "vegetarian": false,
            }
          }
          toggleModal={null}
          type="customer"
        />
        <MealCard
          key="36d525d1-efc9-4b75-9999-3e3d8dc64ce3"
          meal={
            Object {
              "description": "Sharwama contains no animal products, perfect for dieters",
              "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
              "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
              "price": "1200.00",
              "title": "Vegetable Sharwama and Guava Smoothie",
              "vegetarian": true,
            }
          }
          toggleModal={null}
          type="customer"
        />
      </div>
    </React.Fragment>
  </div>
</CustomerView>
`;

exports[`Menu renders message when not fetching and there are no meals on the menu 1`] = `
<CustomerView
  isFetching={true}
  logout={[MockFunction]}
  type="menu"
  user={
    Object {
      "email": "emiola@olasanmi.com",
      "firstname": "Emiola",
      "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
      "lastname": "Olasanmi",
      "role": "customer",
    }
  }
>
  <div
    className="content-wrapper meals user-meals"
  >
    <div
      className="page-heading"
    >
      <h2>
        Today's Menu
      </h2>
      <hr />
    </div>
    <React.Fragment>
      <p
        className="text-center"
      >
        There are no Meals on Today's Menu
      </p>
    </React.Fragment>
  </div>
</CustomerView>
`;
